qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -284,8 +284,65 @@ def observable(me_table, init_term=0, mapping="jordan_wigner"):
     return structure.convert_observable(jordan_wigner(mb_obs))
 
 
+def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
+    r"""Reads the Hartree-Fock (HF) electronic structure data file, defines an active
+    space and generates the table of matrix elements required to build the second-quantized
+    spin-projection operator :math:`\hat{S}_z`.
+
+    The single-particle operator :math:`\hat{s}_z` is diagonal in the basis
+    of spin-orbitals and its matrix elements are computed as follows,
+
+    .. math::
+
+        \langle \alpha \vert \hat{s}_z \vert \beta \rangle = m_\alpha \delta_{\alpha,\beta},
+
+    where :math:`m_\alpha` refers to the quantum number of the spin wave function
+    :math:`\chi_{m_\alpha}(s_z)` of the spin-orbital :math:`\vert \alpha \rangle`.
+
+    Args:
+        mol_name (str): name of the molecule
+        hf_data (str): path to the directory with the HF electronic structure data
+        n_active_electrons (int): number of active electrons
+        n_active_orbitals (int): number of active orbitals
+
+    Returns:
+        array: NumPy array with the table of matrix elements. The first two columns
+        contains the index :math:`\alpha` and the third column stores the computed
+        (diagonal) matrix element.
+
+    **Example**
+
+    >>> get_spinZ_matrix_elements(
+        'h2',
+        './pyscf/sto-3g',
+        n_active_electrons=2,
+        n_active_orbitals=2)
+    [[ 0.   0.   0.5]
+    [ 1.   1.  -0.5]
+    [ 2.   2.   0.5]
+    [ 3.   3.  -0.5]]
+    """
+
+    active_indices = structure.active_space(
+        mol_name,
+        hf_data,
+        n_active_electrons=n_active_electrons,
+        n_active_orbitals=n_active_orbitals,
+    )[1]
+
+    n_spin_orbs = 2 * len(active_indices)
+    sz = np.where(np.arange(n_spin_orbs) % 2 == 0, 0.5, -0.5)
+
+    spinz_matrix_elements = np.zeros((n_spin_orbs, 3))
+    for alpha in range(n_spin_orbs):
+        spinz_matrix_elements[alpha] = np.array([alpha, alpha, sz[alpha]])
+
+    return spinz_matrix_elements
+
+
 __all__ = [
     "spin2_matrix_elements",
     "get_spin2_matrix_elements",
     "observable",
+    "get_spinZ_matrix_elements",
```

Comment:
How come `spin2_matrix_elements` exists, but not `spinZ_matrix_elements`?

Reply:
As I commented above, for relatively simple observable we don't need to have a separate function to compute the matrix elements. Another possibility is to get rid of ``spin2_matrix_element`` and fuse it with ``get_spin2_matrix_element``. However, I suggest to have a separate function to more complicated matrix elements.

Reply:
Seems like it's best to make `spin_matrix_element` a private function

Reply:
what do you think @josh146 ?

Reply:
Yes, I think that makes sense. My suggestions are probably:

* Make the helper function private: `_spin2_matrix_elements` 
* Instead of 'get', use 'load', as this typically indicates that a file on disk will be read: `load_spin2_matrix_elements`. But up to you, feel free on this one!